doc/source/tutorial/csgraph.rst

```diff
@@ -47,9 +47,9 @@ later::
     >>> word_list = [word for word in word_list if len(word) == 3]
     >>> word_list = [word for word in word_list if word[0].islower()]
     >>> word_list = [word for word in word_list if word.isalpha()]
-    >>> word_list = map(str.lower, word_list)
+    >>> word_list = list(map(str.lower, word_list))
```

Comment:
The main issue here is that there are unicode characters in the travis word list. e.g.

```
>>> 'aèr'.encode('ascii')
UnicodeEncodeError: 'ascii' codec can't encode character '\xe8' in position 1: ordinal not in range(128)
>>> len('aèr'.encode('utf-8'))
4
>>> len('aer'.encode('utf-8'))
3
```

So it's not possible to encode to `ascii` unless one ignores those characters. It's not a good idea to encode to `UTF-8` because different words require different numbers of bytes:
```
>>> word_list = [word.encode('UTF-8') for word in ['aèr', 'aer', 'èèè']]
>>> word_list = np.asarray(word_list)
>>> word_list.dtype
dtype('S6')
>>> print(word_list)
[b'a\xc3\xa8r' b'aer' b'\xc3\xa8\xc3\xa8\xc3\xa8']
>>> word_bytes = np.ndarray((len(word_list), word_list.itemsize),
...                                                dtype='uint8',
...                                                buffer=word_list.data)
>>> print(word_bytes)
array([[ 97, 195, 168, 114,   0,   0],
       [ 97, 101, 114,   0,   0,   0],
       [195, 168, 195, 168, 195, 168]], dtype=uint8)
```
The words that have no 'fancy characters' only take three bytes and the rest of the row is padded zero. The total number of columns in `word_bytes` depends on the word that has the largest number of bytes in its UTF-8 representation.

 The way that the PR works at the moment is as follows:

- Each of the 3 letter words in `word_list` is UTF-8, that's the default encoding in my locale.
- `word_list = np.asarray(word_list)` puts them into a numpy array which holds 3 unicode characters. `word_list.itemsize` is 12, a UTF-8 character has a maximum size of 4 bytes.
-   In the conversion from `word_list` to `word_bytes` we have to reserve 12 bytes for each word (number of columns).
```
 >>> word_bytes = np.ndarray((word_list.size, word_list.itemsize),
    ...                         dtype='uint8',
    ...                         buffer=word_list.data)
```
- I'm assuming that all characters can be represented by single bytes, so only the first byte of every character is required, so to represent each letter take every 4th byte: `word_bytes = word_bytes[:, ::4]`, which is equal to `word_bytes[:, ::word_list.itemsize // 3`. This approach allows you to create a byte array for each of `['aèr', 'aer', 'èèè']`